Make errors

Im on CentOS 6.5 x86_64. Fresh installation and fully updated.
Im getting this at the end of the make.

zone_data.cc: In member function ‘void bundy::datasrc::memory::ZoneData::removeNode(bundy::util::MemorySegment&, bundy::datasrc::memory::ZoneNode_)’:
zone_data.cc:265: error: no match for ‘operator==’ in ‘node == ((bundy::datasrc::memory::ZoneData_)this)->bundy::datasrc::memory::ZoneData::origin_node_’
make[8]: **\* [zone_data.lo] Error 1
make[7]: **\* [all-recursive] Error 1
make[6]: **\* [all] Error 2
make[5]: **\* [all-recursive] Error 1
make[4]: **\* [all] Error 2
make[3]: **\* [all-recursive] Error 1
make[2]: **\* [all-recursive] Error 1
make[1]: **\* [all-recursive] Error 1
make: **\* [all] Error 2

Configure command:
[root@dns1 bundy]# ./configure --prefix=/opt/bundy10 --with-pythonpath=/opt/python3/bin/python3  --with-botan-config=/opt/botan/bin/botan-config-1.10 --with-log4cplus=/opt/log4cplus/ --without-shared-memory

--- a/src/lib/datasrc/memory/zone_data_loader.cc
+++ b/src/lib/datasrc/memory/zone_data_loader.cc
@@ -523,7 +523,7 @@ ZoneDataLoader::ZoneDataLoaderImpl::doLoadCommon(
 
             // Check loaded serial.  Note that if checkZone() passed, we
             // should have SOA in the ZoneData.
-            boost::scoped_ptr<const dns::Serial>new_serial(
+            boost::scoped_ptr<const dns::Serial> new_serial(
                 getSerialFromZoneData(rrclass_, loaded_data));
             if (old_serial_ && *old_serial_ >= *new_serial) {
                 LOG_WARN(logger, DATASRC_MEMORY_LOADED_SERIAL_NOT_INCREASED).

